game-core/src/main/java/games/strategy/triplea/ui/MapRouteDrawer.java

```diff
@@ -314,28 +323,81 @@ private void drawCurvedPath(final Graphics2D graphics, final Point2D[] points) {
   }
 
   /**
-   * This draws how many moves are left on the given {@linkplain BufferedImage}
-   *
-   * @param image The Image to be drawn on
-   * @param curMovement How many territories the unit traveled so far
-   * @param maxMovement How many territories is allowed to travel. Is empty when the unit traveled too far
+   * This draws current moves, max moves, and fuel cost.
    */
-  private static void createMovementLeftImage(final BufferedImage image, final String curMovement,
-      final String maxMovement) {
-    final Graphics2D textG2D = image.createGraphics();
-    textG2D.setColor(Color.YELLOW);
-    textG2D.setFont(new Font("Dialog", Font.BOLD, 20));
-    final int textThicknessOffset = textG2D.getFontMetrics().stringWidth(curMovement) / 2;
-    final boolean distanceTooBig = maxMovement.isEmpty();
-    textG2D.drawString(
+  private static BufferedImage createMovementLeftImage(final String curMovement, final String maxMovement,
+      final ResourceCollection movementFuelCost, final ResourceImageFactory resourceImageFactory) {
+
+    // Create and configure image
+    final String unitMovementLeft = (maxMovement == null || maxMovement.trim().length() == 0) ? "" : "/" + maxMovement;
+    final int imageWidth = findMovementLeftImageWidth(curMovement, unitMovementLeft, movementFuelCost);
+    final BufferedImage image = new BufferedImage(imageWidth, MESSAGE_HEIGHT, BufferedImage.TYPE_INT_ARGB);
+    final Graphics2D graphics = image.createGraphics();
+    graphics.setRenderingHint(RenderingHints.KEY_ANTIALIASING, RenderingHints.VALUE_ANTIALIAS_ON);
+    graphics.setFont(MESSAGE_FONT);
+    final FontMetrics fontMetrics = graphics.getFontMetrics();
+
+    // Draw background
+    graphics.setColor(new Color(255, 250, 220));
+    final int rectHeight = MESSAGE_HEIGHT - 2;
+    graphics.fillRoundRect(0, 0, imageWidth - 2, rectHeight, rectHeight, rectHeight);
+    graphics.setColor(Color.BLACK);
+    graphics.drawRoundRect(0, 0, imageWidth - 2, rectHeight, rectHeight, rectHeight);
+
+    // Draw current movement
+    graphics.setColor(new Color(255, 120, 0));
```

Comment:
Just noting that I find this foreground color a bit hard to distinguish between the background color.  It looks like it's a medium orange on sand.  Not a big deal, just wanted to mention it.

Reply:
I adjusted a couple of the colors so that its more of a light gray background and a darker orange.